adapters/pubmatic/pubmatic.go

```diff
@@ -550,30 +551,40 @@ func (a *PubmaticAdapter) MakeBids(internalRequest *openrtb.BidRequest, external
 			bid := sb.Bid[i]
 			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
 				Bid:     &bid,
-				BidType: getMediaTypeForImp(bid.ImpID, internalRequest.Imp),
+				BidType: getBidType(bid.Ext),
 			})
 
 		}
 	}
 	return bidResponse, errs
 }
 
-// getMediaTypeForImp figures out which media type this bid is for.
-func getMediaTypeForImp(impId string, imps []openrtb.Imp) openrtb_ext.BidType {
-	mediaType := openrtb_ext.BidTypeBanner
-	for _, imp := range imps {
-		if imp.ID == impId {
-			if imp.Video != nil {
-				mediaType = openrtb_ext.BidTypeVideo
-			} else if imp.Audio != nil {
-				mediaType = openrtb_ext.BidTypeAudio
-			} else if imp.Native != nil {
-				mediaType = openrtb_ext.BidTypeNative
+// getBidType returns the bid type specified in the response bid.ext
+func getBidType(bidExt json.RawMessage) openrtb_ext.BidType {
+	// setting "banner" as the default bid type
+	bidType := openrtb_ext.BidTypeBanner
+	if bidExt != nil {
+		bidExtMap := make(map[string]interface{})
+		extbyte, err := json.Marshal(bidExt)
+		if err == nil {
+			err = json.Unmarshal(extbyte, &bidExtMap)
+			if err == nil && bidExtMap[bidTypeExtKey] != nil {
+				bidTypeVal := int(bidExtMap[bidTypeExtKey].(float64))
+				switch bidTypeVal {
+				case 0:
+					bidType = openrtb_ext.BidTypeBanner
+				case 1:
+					bidType = openrtb_ext.BidTypeVideo
+				case 2:
+					bidType = openrtb_ext.BidTypeNative
```

Comment:
The `parseImpressionObject(imp *openrtb.Imp, wrapExt *string, pubID *string)` function will filter out imps from `request.Imp[]` that are not `Video` nor `Banner`. Therefore, are we sure the `case:2` branch is reachable? Is `BidTypeNative` possible if we make it all the way here?.